app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -220,11 +227,44 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
     const { flattenedRequiredScopes, flattenedOptionalScopes } = processScopes(
       requiredScopes,
       optionalScopes,
-      hooks.findNetworkClientIdByChainId,
+      { findNetworkClientIdByChainId, getInternalAccounts },
     );
+
+    const accounts = [];
+    Object.keys(flattenedRequiredScopes).forEach((scope) => {
+      (flattenedRequiredScopes[scope].accounts || []).forEach(
+        (caipAccountId) => {
+          accounts.push(parseAccountId(caipAccountId).address);
+        },
+      );
+    });
+    Object.keys(flattenedOptionalScopes).forEach((scope) => {
+      (flattenedOptionalScopes[scope].accounts || []).forEach(
+        (caipAccountId) => {
+          accounts.push(parseAccountId(caipAccountId).address);
+        },
+      );
+    });
+
+    if (accounts.length > 0) {
+      await hooks.requestPermissions(
+        { origin },
+        {
+          [RestrictedMethods.eth_accounts]: {
+            caveats: [
+              {
+                type: CaveatTypes.restrictReturnedAccounts,
+                value: unique(accounts),
+              },
+            ],
+          },
+        },
+      );
+    }
```

Comment:
is the idea here that we're just doing this for testing until we add the multichain permission middleware?